charon/src/ast/values_utils.rs

```diff
@@ -303,3 +303,68 @@ impl<'de> Deserialize<'de> for ScalarValue {
         deserializer.deserialize_map(Visitor)
     }
 }
+
+impl Serialize for FloatValue {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let enum_name = "FloatValue";
+        let variant_name = self.variant_name();
+        let (variant_index, _variant_arity) = self.variant_index_arity();
+        use rustc_apfloat::Float;
+        let v = match self {
+            FloatValue::F16(bits) => {
+                let v = rustc_apfloat::ieee::Half::from_bits(*bits);
+                v.to_string()
+            },
+            FloatValue::F32(bits) => {
+                let v = rustc_apfloat::ieee::Single::from_bits(*bits);
+                v.to_string()
+            },
+            FloatValue::F64(bits) => {
+                let v = rustc_apfloat::ieee::Double::from_bits(*bits);
+                v.to_string()
+            },
+            FloatValue::F128(bits) => {
+                let v = rustc_apfloat::ieee::Quad::from_bits(*bits);
+                v.to_string()
+            }
+        };
+        serializer.serialize_newtype_variant(enum_name, variant_index, variant_name, &v)
+    }
+}
+
+impl<'de> Deserialize<'de> for FloatValue {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct Visitor;
+        impl<'de> serde::de::Visitor<'de> for Visitor {
+            type Value = FloatValue;
+            fn expecting(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+                write!(f, "FloatValue")
+            }
+            fn visit_map<A: serde::de::MapAccess<'de>>(
+                self,
+                mut map: A,
+            ) -> Result<Self::Value, A::Error> {
+                use serde::de::Error;
+                let (k, v): (String, String) = map.next_entry()?.expect("Malformed FloatValue");
+                Ok(match k.as_str() {
+                    "F16" => FloatValue::F16(v.parse().unwrap()),
+                    "F32" => FloatValue::F32(v.parse().unwrap()),
+                    "F64" => FloatValue::F64(v.parse().unwrap()),
+                    "F128" => FloatValue::F128(v.parse().unwrap()),
```

Comment:
This will try to parse a float value into a `u128`, it won't work

Reply:
We're now using automatically derived (de)serializers since we are not risking integer overflows anymore with the new float value representation